tokio/src/io/poll_evented.rs

```diff
@@ -124,6 +124,17 @@ impl<E: Source> PollEvented<E> {
     pub(crate) fn registration(&self) -> &Registration {
         &self.registration
     }
+
+    /// Deregister the inner io from the registration and returns a Result containing the inner io
+    #[cfg(feature = "net")]
+    pub(crate) fn into_inner(mut self) -> io::Result<E> {
+        let mut inner = self
+            .io
+            .take()
+            .ok_or_else(|| io::Error::from(io::ErrorKind::NotFound))?;
```

Comment:
I see that other parts of the code just `unwrap` this. Is it the case that it is an `Option` only for the purpose of being possible to take it out in or before the destructor? @carllerche

Reply:
Yes, a panic here is fine as it would be an internal bug.

Reply:
I believe @Darksonn is correct — this _shouldn't_ ever be `None` unless the `PollEvented` is being dropped or `into_std` has been called, so this can just be an `unwrap()`...